PearlPCSup/devPearl.db

```diff
@@ -169,75 +150,79 @@ record(bo, "$(P)$(IOC_NAME)SL_PRESSURE:SP"){
     field(DESC, "Set toggle for loop pressure mode")
     field(DTYP, "stream")
     field(OUT, "@PearlPC.proto set_sloop $(PORT)")
-    field(ZNAM, "OFF")
-    field(ONAM, "ON")
+    field(ZNAM, "Open Loop")
+    field(ONAM, "Closed Loop")
 }
 
 record(bi, "$(P)$(IOC_NAME)SL_PRESSURE"){
     field(DESC, "Get loop pressure mode")
-    field(INP, "$(P)$(IOC_NAME)BUFFER1.H CP MS")
+    field(INP, "$(P)$(IOC_NAME)STATUS_ARRAY.[7] CP MS")
     field(DTYP, "Soft Channel")
-    field(ZNAM, "OFF")
-    field(ONAM, "ON")
+    field(ZNAM, "Open Loop")
+    field(ONAM, "Closed Loop")
 }
 
-record(ao, "$(P)$(IOC_NAME)SF_PRESSURE:SP"){
+record(longout, "$(P)$(IOC_NAME)SF_PRESSURE:SP"){
     field(DESC, "Set Seal Fail Pressure Value")
     field(DTYP, "stream")
     field(OUT, "@PearlPC.proto set_sf $(PORT)")
 }
 
-record(ai, "$(P)$(IOC_NAME)SF_PRESSURE"){
-    field(DESC, "Get Seal Fail Pressure")
-    field(INP, "$(P)$(IOC_NAME)BUFFER1.I CP MS")
+# this would appear to be a binary of wther seal has failed not the SF setpoint from above?
```

Comment:
I don't think  `STATUS_ARRAY.[8]` is the pressure value, hence my confusion with the logic. From the manual it says it is `sf	Seal fail 1 = sudden pressure drop detected` so it tells you there is a seal fail i guess based on the current seal fail pressure set? To do the logic you have you would need to read this pressure value by another route,  does the labview do this check as it would appear that the controller is already doing this and returning the result in the STAUS word?

Reply:
So just to clarify that 8 is the Seal Fail Pressure? If I remember correctly and my understanding of the below logic is correct, Once SF_PRESSURE is set by 8 (representing the seal pressure failure drop value), it forward links to `DETECT_SEAL_FAIL` which then detects when drops in pressure larger than the `SF_PRESSURE` value occurs, placing the controller in seal fail mode when true to `SF_MODE_PRESSURE`. The below functionality assumes that the logic is not handled automatically by the controller. Once we get the hardware to test this on, we should be able to clear up whether or not this needs to stay based on whether the logic is handled by the controller. I didn't understand the LabVIEW well enough to determine what is going on there.

Reply:
My reading is that SF_PRESSURE (as in STATUS_ARRAY.[8]) will only ever be 0 or 1, with 1 indicating that it has detected a seal fail with a pressure drop larger than the value previously specified to the controller via the `sf` command.  But we can check that with labview/hardware